packages/core/src/common/keybinding.ts

```diff
@@ -267,4 +303,37 @@ export class KeybindingRegistry {
     isPseudoCommand(commandId: string): boolean {
         return commandId === KeybindingRegistry.PASSTHROUGH_PSEUDO_COMMAND;
     }
+
+    setKeymap(scope: number, rawKeyBindings: RawKeybinding[]) {
+        const customBindings: Keybinding[] = [];
+        for (const rawKeyBinding of rawKeyBindings) {
+            if (this.commandRegistry.getCommand(rawKeyBinding.command)) {
+                const code = KeyCode.parse(rawKeyBinding.keybinding);
+                if (code) {
+                    let context: KeybindingContext | undefined;
+                    if (rawKeyBinding.context) {
+                        context = this.contextRegistry.getContext(rawKeyBinding.context);
+                    }
+
+                    customBindings.push({
+                        commandId: rawKeyBinding.command,
+                        keyCode: code,
+                        contextId: context ? context.id : undefined
+                    });
+
+                } else {
+                    this.resetKeybindings(scope);
+                    return;
+                }
+            } else {
+                this.resetKeybindings(scope);
+                return;
+            }
+        }
+        this.keymaps[scope] = customBindings;
+    }
+
+    resetKeybindings(scope: number) {
```

Comment:
ForScope? seems like you could have a reset all keybindings function.. that would be called resetKeybindings ?